core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -506,7 +516,16 @@ private[spark] class AppStatusListener(
       if (killedDelta > 0) {
         stage.killedSummary = killedTasksSummary(event.reason, stage.killedSummary)
       }
-      maybeUpdate(stage, now)
+      // Remove stage if there are no active tasks left and stage is already finished
+      val removeStage =
+        stage.activeTasks == 0 &&
+          (v1.StageStatus.COMPLETE.equals(stage.status) ||
+            v1.StageStatus.FAILED.equals(stage.status))
+      if (removeStage) {
+        update(stage, now, last = true)
```

Comment:
Shouldn't this be removing the stage from the live list too?

Reply:
You are correct. I am removing the stage from live list towards the end of this function.